includes/forms/class-llms-form-field.php

```diff
@@ -58,29 +58,57 @@ class LLMS_Form_Field {
 	protected $html = '';
 
 	/**
-	 * Constructor.
+	 * Constructor
 	 *
 	 * @since [version]
 	 *
-	 * @param array $settings Field settings.
-	 *
+	 * @param array      $settings    Field settings.
+	 * @param int|object $data_source Optional. Data source where to get field value from. Default is `null`.
+	 *                                Can be a WP_User or a WP_Post, or their id.
+	 *                                The actual object will be retrieved basing on the data_store.
 	 * @return void
 	 */
-	public function __construct( $settings = array() ) {
+	public function __construct( $settings = array(), $data_source = null ) {
 
 		/**
-		 * Filters the settings of a LifterLMS Form Field.
+		 * Filters the settings of a LifterLMS Form Field
 		 *
 		 * @since [version]
 		 *
 		 * @param array           $settings Field settings.
 		 * @param LLMS_Form_Field $field    Form field class instance.
 		 */
 		$this->settings = apply_filters( 'llms_field_settings', wp_parse_args( $settings, $this->get_defaults() ), $this );
+
+		$this->define_data_source( $data_source );
+
 		$this->prepare();
 
 	}
 
+	/**
+	 * Define the source of the data
+	 *
+	 * @since [version]
+	 *
+	 * @param int|object $data_source Data source where to get field value from.
+	 * @return void
+	 */
+	private function define_data_source( $data_source ) {
+
+		if ( ! empty( $this->settings['data_store'] ) && ! in_array( $this->settings['data_store'], array( 'users', 'usermeta' ), true ) ) {
+			return;
+		}
+		if ( ! is_null( $data_source ) ) {
+			$data_source = $data_source instanceof WP_User ? $data_source : get_user_by( 'ID', $data_source );
+		} elseif ( is_user_logged_in() ) {
+			$data_source = wp_get_current_user();
+		}
+
+		$this->user = $data_source;
+
+	}
+
```

Comment:
I think I'd like to see `$this->user` converted to `$this->data_source`, it's a semantic change but I think it'll be better for us in the future when we add other data sources

Reply:
I don't agree because then we would need to add an additional check on the nature of `$data_source`, instead I want to be sure that $this->user is either unset or it's a WP_User.
Then we'll have `$this->post`.
What do you think? Am I stupid?

Reply:
Yes, you're right... but we'll have to run an additional check either way, won't we, in order to determine the data source being used? EG:

`if ( ! is_null( $this->post ) && is_a( WP_User, $this->user ) )`, right?

If we use `$this->data_source` we can infer the type of data source from the already-existing `data_store` settings.

Although, maybe we should do as you're suggesting or...

What if we add a `$this->data_source_type` (also set in the `define_data_source()` method which would be `user` or `post` depending on the context.

Then we can check against the type later?

Maybe you're right though...

Reply:
> Yes, you're right... but we'll have to run an additional check either way, won't we, in order to determine the data source being used? EG:
> 
> if ( ! is_null( $this->post ) && is_a( WP_User, $this->user ) ), right?
> 
> If we use $this->data_source we can infer the type of data source from the already-existing data_store settings.

well my idea was just to check `if ( ! isset( $this->post ) )` throughout the code, it can be set or not and that should suffice.
But I think I see your "abstraction" point and will go with `$this->data_source_type` and `$this->data_source`.